runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -454,6 +454,17 @@ protected DataflowRunner(DataflowPipelineOptions options) {
     this.ptransformViewsWithNonDeterministicKeyCoders = new HashSet<>();
   }
 
+  /** For portable jobs, Dataflow still requires an override of the PubsubIO transforms. */
+  private List<PTransformOverride> getPortableOverrides() {
+    return ImmutableList.of(
+        PTransformOverride.of(
```

Comment:
We should also check `hasExperiment(options, "enable_custom_pubsub_source")`

Reply:
Yes, good catch.